client/src/main/java/org/apache/rocketmq/client/producer/RequestResponseFuture.java

```diff
@@ -39,7 +41,7 @@ public RequestResponseFuture(String correlationId, long timeoutMillis, RequestCa
     }
 
     public void executeRequestCallback() {
-        if (requestCallback != null) {
+        if (requestCallback != null && executeCallbackOnlyOnce.compareAndSet(false, true)) {
```

Comment:
code has change. see latest code: `if (sendRequestOk && cause == null) {`

Reply:
I have pulled the latest code from 'develop' branch and not very sure what you mean here, could you plz give me more details?